controllers/cluster_controller.go

```diff
@@ -1071,22 +1070,22 @@ func (r *ClusterReconciler) markPVCReadyForCompletedJobs(
 	for _, job := range completeJobs {
 		var pvcName string
 		for _, pvc := range resources.pvcs.Items {
-			if specs.IsJobOperatingOnPVC(job, pvc) {
+			if specs.IsWorkingOnPVC(job.Spec.Template.Spec, pvc.Name) {
 				pvcName = pvc.Name
 				break
 			}
 		}
+
 		if pvcName == "" {
 			continue
 		}
 
 		// finding the PVC having the same name as pod
+		// TODO: we need to do a null check
```

Comment:
Yep. Totally agree with it. Let's make this function return an error and improve the caller too. From what I can see it's called just one time.